flang/test/Lower/intrinsic-procedures.f90

```diff
@@ -717,6 +717,21 @@ subroutine modulo_testi(r, a, p)
   r = modulo(a, p)
 end subroutine
 
+! NOT
+! CHECK-LABEL: not_test
+subroutine not_test
+  integer :: source
+  integer :: destination
+  ! CHECK:  %0 = fir.alloca i32 {bindc_name = "destination", uniq_name = "_QFnot_testEdestination"}
```

Comment:
In tests, we avoid to expect precise SSA value indexes (the `%xxx`s ). Instead, we use FileCheck pattern matching to capture and verify SSA value usages without depending on the number.
This can be done by replacing `%xxx = ` by `%[[some_name:.*]] =`, and by replacing later use if `%xxx` by `%[[some_name]]`

For example:
```
  ! CHECK: %[[dest:.*]] = fir.alloca i32 {bindc_name = "destination", uniq_name = "_QFnot_testEdestination"
  ! CHECK:  %[[source:.*]] = fir.alloca i32 {bindc_name = "source", uniq_name = "_QFnot_testEsource"}
  ! CHECK:  %[[load:.*]] = fir.load %[[source]] : !fir.ref<i32>
  ! ....
```

It is quite annoying to write, but avoids having to maintain too much the tests every-time there is a small unrelated change that change the ssa number indexes.

You can run `LLVM_SOURCES/mlir/utils/generate-test-checks.py` on a file containing your expected FIR output to get all the pattern matching inserted automatically. For small tests, it's quite convenient, for bigger ones also, though if a lot of FIR is generated, it's better to only focus on the few operations rather than to test the whole output.

Reply:
Thanks for the great comments.  I've submitted corrections.